app/(root)/page.tsx

```
@@ -1,27 +1,99 @@
-// 1. izvaditi sve boje iz figme u tailwind config
-// 2. namestiti font-family
-// 3. namestiti sve variants button-a
-// * kada namestim config i button as example
-// 1. MongoDB --- nov projekat
-// 2. mongoose --- namestiti povezati
-// 3. Next-auth --- email pw, google, github
-// 4. Login page --- sign-up i sign-in ( prvo uraditi sa github-om pa google)
-
-// * branch
-// feature/tailwind-styles
-// feat / tailwind - styling;
-// feat / authentication;
-
-// app/(auth)/login/page.tsx implement RHF + ZOD ili Yup
-// REGIISTE
-const Home: React.FC = () => {
-  // fetch all post
+import type { HeatMapValue } from '@uiw/react-heat-map';
+
+import { auth } from '@/auth';
+import PostItem from '@/components/post/PostItem';
+import PostItemBadge from '@/components/post/PostItemBadge';
+import Pagination from '@/components/shared/Pagination';
+import PostsHeatMap from '@/components/shared/PostsHeatMap';
+import { POST_TYPES } from '@/constants';
+import { getAllPosts, getHeatMapPostsData } from '@/lib/actions/post-actions';
+import type { IPostsResponse } from '@/types/post';
+import { EQueryPostType } from '@/types/post-types';
+import { parseSearchParams, parseTagsParams } from '@/utils/params';
+
+// ----------------------------------------------------------------
+
+/**
+ * pogledati tiny mce custom plugin za Warning code message
+ * react calendar heatmap ===> pratimo broj commitova pushova na / HOME
+ */
+
+interface IHomeProps {
+  searchParams: {
+    page: string | string[] | undefined;
+    postType: string | string[] | undefined;
+    tag: string | string[] | undefined;
+  };
+}
+
+const Home: React.FC<IHomeProps> = async ({ searchParams }) => {
+  const page = parseSearchParams(searchParams.page, '1');
+  const postType = parseSearchParams(
+    searchParams.postType,
+    ''
+  ) as EQueryPostType;
+  const tags = parseTagsParams(searchParams.tag);
+
+  const session = await auth();
+
+  if (!session) throw new Error('User data is not available!');
```

Check if the middleware file is protecting this route. We only want to throw an error if this is a protected route, the middleware let's the user through to the route, and for some reason session ends up undefined. Just check that you're not using this to protect your route, it's okay to throw an error if this error really happened, not if you're just trying to protect a route.